avgsamplerate.go

```diff
@@ -25,6 +25,10 @@ type AvgSampleRate struct {
 	// events. Default 10
 	GoalSampleRate int
 
+	// MaxKeys, if greater than 0, limits the number of distinct keys used to build
+	// the sample rate map.
```

Comment:
I would love some color here describing what happens when max keys is hit. Do new keys seen replace existing keys? Are all new keys ignored once the limit is hit? Do existing keys continue to get counted? Since this comment shows up in godocs it's a good place to add some detail about the implementation and its effects.